### PR TITLE
feat: 캐러셀 통일 및 쿼리 추가

### LINKED ISSUE #90: [REFACTOR] 컴포넌트 수정 및 에러 핸들링
Screen Code: 공통 컴포넌트 작업 수정 및 에러 핸들링
Due to: 2024-03-25
Description:
- 공통 컴포넌트 작업
- [x] 이미지 리스트 컴포넌트
- [ ] 캐러셀 swiper 라이브러리 사용한 Banner, Carousel 통일화
- [x] Toast, Alert, Modal 컴포넌트 추가
- API 에 대한 에러 핸들링
- [ ] api 에러에 대한 공통 errorHandler 작성
- [ ] ErrorBoundary 설정

### PR DESCRIPTION
**Pull request by sy-paik**
**What is this PR?**
- 불필요한 라이브러리를 삭제하고 코드 통일화 작업을 진행하였습니다.

**What has changed?**
- react query -> tanstack/react-query로 변경 (최신 버전)
- api 쿼리 hook으로 분리: 각 페이지 중심으로 api 훅을 작성하였습니다 
- API 요청을 수행하는 HTTP 동작을 각각의 함수로 만들어 mutationFn과 queryFn에서 사용하였습니다.
- 캐러셀이 메인과 총학생회 각 페이지에 분리되어 있어 이를 통일화하였습니다. (메인의 캐러셀 타입에서 총학생회의 캐러셀 타입만 확장하여 타입을 통일화하였습니다,)
- 회원가입에서 닉네임 유효성을 추가하고 성공여부에 따른 메시지가 렌더링되도록 추가하였습니다.
- router의 path는 as const 를 사용하였습니다 -> 값을 불변하게 만들어 타입 안정성을 위함
- intersection 하는 공통 훅과 intersectionBox를 사용하여 각 무한스크롤이 사용되는 쿼리훅은 hooks >api에 useInfiniteQuery를 사용하였습니다 -> 각 쿼리에 따른 Response type이 다르기 때문
- 재사용성이 높은 타입은 types에 정의해두어 필요한 곳에 import 하여 사용하도록 변경하였습니다.

**A notice to reviewers...**
- 에러바운더리는 다음 이슈에서 추가적으로 올리도록 하겠습니다..! (이 부분도 변경된 파일이 많아질 것 같네요ㅠ) 아직 에러바운더리를 어떤식으로 잡을지는 결정하지 못했는데 편하게 의견주셔도 좋습니다! 
- 추가하지 못한 쿼리 훅들 추가하다가 너무 파일이 많아졌는데 죄송합니다,.. 

closes #90

### CODE REVIEW

**Comment by gaeunnlee** [comment]
오류 확인해주실 수 있을까요?🥲

ERROR in ./src/libs/api.ts 32:54-68
export 'handleAPIError' (imported as 'handleAPIError') was not found in '@libs/interceptor' (possible exports: authorization, checkToken)

ERROR in ./src/pages/mypage/index.tsx 26:4-16
export 'removeTokens' (imported as 'removeTokens') was not found in '@/utils/token' (possible exports: getAccessToken, getRefreshToken, isLoggedIn, removeToken, setToken)

ERROR
[eslint] 
src\components\ui\toast\use-toaster.ts
  Line 105:1:  Expected indentation of 21 spaces but found 23  indent
  Line 106:1:  Expected indentation of 21 spaces but found 23  indent
  Line 107:1:  Expected indentation of 18 spaces but found 20  indent

Search for the keywords to learn more about each error.

ERROR in ./src/libs/api.ts 32:54-68
export 'handleAPIError' (imported as 'handleAPIError') was not found in '@libs/interceptor' (possible exports: authorization, checkToken)

ERROR in ./src/pages/mypage/index.tsx 26:4-16
export 'removeTokens' (imported as 'removeTokens') was not found in '@/utils/token' (possible exports: getAccessToken, getRefreshToken, isLoggedIn, removeToken, setToken)
ERROR
[eslint] 
src\components\ui\toast\use-toaster.ts
  Line 105:1:  Expected indentation of 21 spaces but found 23  indent
  Line 106:1:  Expected indentation of 21 spaces but found 23  indent
  Line 107:1:  Expected indentation of 18 spaces but found 20  indent

Search for the keywords to learn more about each error.

ERROR in src/components/signup/info.tsx:28:41
TS2339: Property 'response' does not exist on type '{}'.
    26 |    } = usePostPhoneConfirmCode(signupToken);
    27 |    const { mutate: signup } = usePostSignup(signupToken);
  > 28 |    const codeErrorResponse = codeError?.response?.data as HTTPError;
       |                                         ^^^^^^^^
    29 |    const { isSuccess: isNicknameVerify, refetch } = useGetNicknameVerify(signupInfo.nickname);
    30 |
    31 |    const labelStyle = 'ml-[14px] font-normal text-gray02';

ERROR in src/libs/api.ts:2:10
TS2305: Module '"@libs/interceptor"' has no exported member 'handleAPIError'.
    1 | import { CONSTANTS } from '@constants/api';
  > 2 | import { handleAPIError } from '@libs/interceptor';
      |          ^^^^^^^^^^^^^^
    3 | import axios, { AxiosResponse } from 'axios';
    4 |
    5 | export const client = axios.create({

ERROR in src/pages/mypage/index.tsx:11:10
TS2724: '"@/utils/token"' has no exported member named 'removeTokens'. Did you mean 'removeToken'?
     9 |
    10 | import MyPageLayout from '@/components/layouts/MyPageLayout';
  > 11 | import { removeTokens } from '@/utils/token';
       |          ^^^^^^^^^^^^
    12 |
    13 | export default function MyPage() {
    14 |    const navigate = useNavigate();

**Comment by sy-paik** [comment]
@gaeunnlee 너무 늦어져서 죄송합니다.. ㅠ useMutation이나 useQuery에서 에러 발생 시, error가 null이 떠서 이부분 해결하느라 더 오래걸렸네요.. 충돌도 같이 수정했는데 확인부탁드릴게요!

**Comment by sy-paik** [comment]
> 오류 확인해주실 수 있을까요?🥲
> 
> ERROR in ./src/libs/api.ts 32:54-68 export 'handleAPIError' (imported as 'handleAPIError') was not found in '@libs/interceptor' (possible exports: authorization, checkToken)
> 
> ERROR in ./src/pages/mypage/index.tsx 26:4-16 export 'removeTokens' (imported as 'removeTokens') was not found in '@/utils/token' (possible exports: getAccessToken, getRefreshToken, isLoggedIn, removeToken, setToken)
> 
> ERROR [eslint] src\components\ui\toast\use-toaster.ts Line 105:1: Expected indentation of 21 spaces but found 23 indent Line 106:1: Expected indentation of 21 spaces but found 23 indent Line 107:1: Expected indentation of 18 spaces but found 20 indent
> 
> Search for the keywords to learn more about each error.
> 
> ERROR in ./src/libs/api.ts 32:54-68 export 'handleAPIError' (imported as 'handleAPIError') was not found in '@libs/interceptor' (possible exports: authorization, checkToken)
> 
> ERROR in ./src/pages/mypage/index.tsx 26:4-16 export 'removeTokens' (imported as 'removeTokens') was not found in '@/utils/token' (possible exports: getAccessToken, getRefreshToken, isLoggedIn, removeToken, setToken) ERROR [eslint] src\components\ui\toast\use-toaster.ts Line 105:1: Expected indentation of 21 spaces but found 23 indent Line 106:1: Expected indentation of 21 spaces but found 23 indent Line 107:1: Expected indentation of 18 spaces but found 20 indent
> 
> Search for the keywords to learn more about each error.
> 
> ERROR in src/components/signup/info.tsx:28:41 TS2339: Property 'response' does not exist on type '{}'. 26 | } = usePostPhoneConfirmCode(signupToken); 27 | const { mutate: signup } = usePostSignup(signupToken);
> 
> > 28 |    const codeErrorResponse = codeError?.response?.data as HTTPError;
> > |                                         ^^^^^^^^
> > 29 |    const { isSuccess: isNicknameVerify, refetch } = useGetNicknameVerify(signupInfo.nickname);
> > 30 |
> > 31 |    const labelStyle = 'ml-[14px] font-normal text-gray02';
> 
> ERROR in src/libs/api.ts:2:10 TS2305: Module '"@libs/interceptor"' has no exported member 'handleAPIError'. 1 | import { CONSTANTS } from '@constants/api';
> 
> > 2 | import { handleAPIError } from '@libs/interceptor';
> > |          ^^^^^^^^^^^^^^
> > 3 | import axios, { AxiosResponse } from 'axios';
> > 4 |
> > 5 | export const client = axios.create({
> 
> ERROR in src/pages/mypage/index.tsx:11:10 TS2724: '"@/utils/token"' has no exported member named 'removeTokens'. Did you mean 'removeToken'? 9 | 10 | import MyPageLayout from '@/components/layouts/MyPageLayout';
> 
> > 11 | import { removeTokens } from '@/utils/token';
> > |          ^^^^^^^^^^^^
> > 12 |
> > 13 | export default function MyPage() {
> > 14 |    const navigate = useNavigate();

너무 늦어져서 죄송합니다.. ㅠ useMutation이나 useQuery에서 에러 발생 시, error가 null이 떠서 이부분 해결하느라 더 오래걸렸네요.. 충돌도 같이 수정했는데 확인부탁드릴게요!

File: src/hooks/api/petition/useGetPetiition.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+   agreeCount: number;
+}
+
+interface PetitionResponse extends PageResponse {
+   content: PetitionContentResponse[];
+}
+
+export const useGetPetition = () => {
+   return useInfiniteQuery<PetitionResponse>({
+      queryKey: ['getRule'],
```
Comment (gaeunnlee) [inline src/hooks/api/petition/useGetPetiition.ts]: 키 입력이 잘못된 것 같습니다!

**Review by gaeunnlee — APPROVED** [review]
수고 많으셨습니다!🥹

**Pull request merged**